compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7684,14 +7684,9 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
             restType = data.resultType;
         } else if (vararg != null) {
             iExpr.restArgs.add(vararg);
+            checkTypeParamExpr(vararg, listTypeRestArg, iExpr.langLibInvocation, data);
             if (mappingTypeRestArg != null) {
-                LinkedHashSet<BType> restTypes = new LinkedHashSet<>();
-                restTypes.add(listTypeRestArg);
-                restTypes.add(mappingTypeRestArg);
-                BType actualType = BUnionType.create(null, restTypes);
-                checkTypeParamExpr(vararg, actualType, iExpr.langLibInvocation, data);
-            } else {
-                checkTypeParamExpr(vararg, listTypeRestArg, iExpr.langLibInvocation, data);
+                checkTypeParamExpr(vararg, mappingTypeRestArg, iExpr.langLibInvocation, data);
```

Comment:
As I understand, this change is to avoid creating the union type because, in the type param, the union case is not handled when the actual type is union. I think it is better to fix the issue in type param phase (the expected type is union and actual type is not union). Let's discuss with others too.